admin/services/data-access/check-window.data.service.js

```diff
@@ -326,6 +326,18 @@ const checkWindowDataService = {
     WHERE isDeleted=0
     ORDER BY name ASC`
     return sqlService.query(sql)
+  },
+
+  /**
+   * Fetch active check window
+   * @return {Object}
+   */
+  sqlFindActiveCheckWindow: async () => {
+    const sql = `SELECT TOP 1 *
```

Comment:
This is better in that it will only return one or zero rows, but it's still possible for this query to return the wrong rows by omission if there are multiple check windows active at the same time.

It ought to be possible to close this off at the database level by ensuring that any new `checkWindow` must NOT overlap any existing `checkWindow`.

This could be done by a trigger to detect overlaps perhaps?

```
select
	original.name,
	original.checkStartDate,
	original.checkEndDate,
	overlap.name,
	overlap.checkStartDate,
	overlap.checkEndDate
	
from
	[mtc_admin].[checkWindow] as original 
	inner join [mtc_admin].[checkWindow] as overlap ON (original.id <> overlap.id)
 
where

  -- overlapping, not just touching
  original.checkEndDate > overlap.checkStartDate AND
  overlap.checkEndDate > original.checkStartDate AND

  -- not exact duplicate
  original.checkEndDate <> overlap.checkEndDate AND
  original.checkStartDate <> overlap.checkStartDate
 
 group by 
 	original.name,
	original.checkStartDate,
	original.checkEndDate,
	overlap.name,
	overlap.checkStartDate,
	overlap.checkEndDate
;
```

Reply:
I would prefer this as a business rule in code.

George - if i recall correctly there are existing rules around this?

Reply:
Not existing but there is a set of rules to-be-implemented to cover this from the service manager perspective.

Reply:
yes, as agreed on standup.  Thanks George